File: 3779.minimum-number-of-operations-to-have-distinct-elements.py
#
# @lc app=leetcode id=3779 lang=python3
#
# [3779] Minimum Number of Operations to Have Distinct Elements
#

# @lc code=start
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        operations = 0
        start = 0
        
        while start < len(nums):
            # Check if remaining elements are all distinct
            remaining = nums[start:]
            if len(remaining) == len(set(remaining)):
                break
            
            # Remove first 3 elements (or all if fewer than 3)
            start += min(3, len(nums) - start)
            operations += 1
        
        return operations
# @lc code=end

File: 3779.minimum-number-of-operations-to-have-distinct-elements.cpp
#
# @lc app=leetcode id=3779 lang=cpp
#
# [3779] Minimum Number of Operations to Have Distinct Elements
#

# @lc code=start
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int operations = 0;
        int start = 0;
        int n = nums.size();
        
        while (start < n) {
            // Check if remaining elements are distinct
            unordered_set<int> seen;
            bool hasDuplicate = false;
            for (int i = start; i < n; i++) {
                if (seen.count(nums[i])) {
                    hasDuplicate = true;
                    break;
                }
                seen.insert(nums[i]);
            }
            
            if (!hasDuplicate) {
                return operations;
            }
            
            // Remove first 3 elements (or all remaining)
            start += min(3, n - start);
            operations++;
        }
        
        return operations;
    }
};
# @lc code=end